### PR TITLE
docs: document apmRequest config

### PR DESCRIPTION
**Pull request by jahtalab**
Provides documentation for #1018

### CODE REVIEW

File: docs/configuration.asciidoc [hunk 1]
```diff
@@ -311,3 +311,69 @@ NOTE: Spans that are captured as part of the ignored transactions would also be
...
+* *Default:* `null`
+
+[source,js]
+----
+apm.init({ apmRequest: (requestSpec) => true})
+----
+
+Arguments:
+
+* `requestSpec` - This is an object that contains the APM http request details:
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc:330]:
```suggestion
* `requestParams` - This is an object that contains the APM http request details:
```

File: docs/configuration.asciidoc [hunk 2]
```diff
@@ -311,3 +311,69 @@ NOTE: Spans that are captured as part of the ignored transactions would also be
...
+
+** `url` - The full url of the APM server
+
+** `method` - Method of the http request
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc:334]: uppercase `HTTP` ?

File: docs/configuration.asciidoc [hunk 3]
```diff
@@ -311,3 +311,69 @@ NOTE: Spans that are captured as part of the ignored transactions would also be
...
+
+`apmRequest` can be used to change or reject requests that are made to the
+APM Server. This config can be set to a function, which is called whenever the agent
+needs to make a request to the APM Server. 
+
+The callback function is called with a single argument and is expected to return 
+an output synchronously. If the return value is `true` then the agent continues 
+with making the (potentially modified) request to the APM server. 
+ 
+If this function returns a falsy value the request is discarded.
```
Comment (vigneshshanmugam) [inline docs/configuration.asciidoc:350]: Add like the `request is discarded and rejected with a error` ? or we can fail silently. Just want to call out the behaviour properly here.

**Review by vigneshshanmugam — APPROVED** [review]

File: docs/configuration.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/configuration.asciidoc:334]: Yes, please 👍

File: docs/configuration.asciidoc [hunk 4]
```diff
@@ -311,3 +311,69 @@ NOTE: Spans that are captured as part of the ignored transactions would also be
...
+
+** `xhr` - The `XMLHttpRequest` instance used by the agent to send the request
+
+`apmRequest` can be used to change or reject requests that are made to the
+APM Server. This config can be set to a function, which is called whenever the agent
+needs to make a request to the APM Server. 
+
+The callback function is called with a single argument and is expected to return 
+an output synchronously. If the return value is `true` then the agent continues 
+with making the (potentially modified) request to the APM server.
```
Comment (bmorelli25) [inline docs/configuration.asciidoc:348]:
```suggestion
with making the (potentially modified) request to the APM Server. 
```

**Review by bmorelli25 — APPROVED** [review]
🚢

**Pull request merged**